x/auth/vesting/types/msgs_test.go

```diff
@@ -79,7 +79,114 @@ func TestPeriodicVestingAccountMsg(t *testing.T) {
 
 	badPeriods := NewMsgCreatePeriodicVestingAccount(fromAddr, toAddr, startTime, []Period{
 		{Length: 0, Amount: amount},
-	})
+	}, false)
 	err = badPeriods.ValidateBasic()
 	require.Error(t, err)
 }
+
+func TestTrueVestingAccountMsg(t *testing.T) {
+	_, _, fromAddr := KeyTestPubAddr()
+	_, _, toAddr := KeyTestPubAddr()
+	amount := NewTestCoins()
+	startTime := int64(100200300)
+	lockupPeriods := []Period{
+		{Length: 200000, Amount: amount},
+	}
+	vestingPeriods := []Period{
+		{Length: 300000, Amount: amount},
+	}
+	msg := NewMsgCreateTrueVestingAccount(fromAddr, toAddr, startTime, lockupPeriods, vestingPeriods, false)
+	route := msg.Route()
+	require.Equal(t, RouterKey, route)
+	tp := msg.Type()
+	require.Equal(t, TypeMsgCreateTrueVestingAccount, tp)
+	err := msg.ValidateBasic()
+	require.Nil(t, err)
+
+	badFromMsg := MsgCreateTrueVestingAccount{
+		FromAddress:    "foo",
+		ToAddress:      toAddr.String(),
+		StartTime:      startTime,
+		LockupPeriods:  lockupPeriods,
+		VestingPeriods: vestingPeriods,
+	}
+	err = badFromMsg.ValidateBasic()
+	require.Error(t, err)
+
+	badToMsg := MsgCreateTrueVestingAccount{
+		FromAddress:    fromAddr.String(),
+		ToAddress:      "foo",
+		StartTime:      startTime,
+		LockupPeriods:  lockupPeriods,
+		VestingPeriods: vestingPeriods,
+	}
+	err = badToMsg.ValidateBasic()
+	require.Error(t, err)
+
+	badPeriods := []Period{{Length: 0, Amount: amount}}
+	badLockup := NewMsgCreateTrueVestingAccount(fromAddr, toAddr, startTime, badPeriods, vestingPeriods, false)
+	err = badLockup.ValidateBasic()
+	require.Error(t, err)
+
+	badVesting := NewMsgCreateTrueVestingAccount(fromAddr, toAddr, startTime, lockupPeriods, badPeriods, false)
+	err = badVesting.ValidateBasic()
+	require.Error(t, err)
+
+	badAmounts := NewMsgCreateTrueVestingAccount(fromAddr, toAddr, startTime, lockupPeriods, []Period{
+		{Length: 17, Amount: amount.Add(amount...)},
+	}, false)
+	err = badAmounts.ValidateBasic()
+	require.Error(t, err)
+
+	emptyPeriods := []Period{}
+	noLockupOk := NewMsgCreateTrueVestingAccount(fromAddr, toAddr, startTime, emptyPeriods, vestingPeriods, false)
+	err = noLockupOk.ValidateBasic()
+	require.Nil(t, err)
+
+	noVestingOk := NewMsgCreateTrueVestingAccount(fromAddr, toAddr, startTime, lockupPeriods, emptyPeriods, false)
+	err = noVestingOk.ValidateBasic()
+	require.Nil(t, err)
+}
+
+func TestClawbackMsg(t *testing.T) {
+	_, _, funderAddr := KeyTestPubAddr()
+	_, _, addr := KeyTestPubAddr()
+	_, _, destAddr := KeyTestPubAddr()
+
+	okMsg := NewMsgClawback(funderAddr, addr, destAddr)
+	route := okMsg.Route()
+	require.Equal(t, RouterKey, route)
+	tp := okMsg.Type()
+	require.Equal(t, TypeMsgClawback, tp)
+	err := okMsg.ValidateBasic()
+	require.Nil(t, err)
```

Comment:
use `require.NoError(t, err)` and `require.Error(t, err)` instead of `require.Nil(t, err)` and `require.NotNil(t, err)`

Reply:
Done, here and elsewhere.